examples/flutter_gallery/lib/demo/bottom_navigation_demo.dart

```diff
@@ -0,0 +1,178 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+class NavigationIconView {
+  NavigationIconView({
+    this.icon,
+    this.title,
+    this.color
+  }) {
+    controller = new AnimationController(
+      duration: kThemeAnimationDuration
+    );
+    _animation = new CurvedAnimation(
+      parent: controller,
+      curve: new Interval(0.5, 1.0, curve: Curves.fastOutSlowIn)
+    );
+  }
+
+  final Icon icon;
+  final Widget title;
+  final Color color;
+  AnimationController controller;
+  CurvedAnimation _animation;
+
+  DestinationLabel destinationLabel() {
+    return new DestinationLabel(
+      icon: icon,
+      title: title,
+      backgroundColor: color
+    );
+  }
+
+  FadeTransition transition(BottomNavigationBarType type, BuildContext context) {
+    Color color;
```

Comment:
Instead of shadowing the color field here, give this a distinctive name. Like viewIconColor.